rasa/nlu/featurizers/sparse_featurizer/count_vectors_featurizer.py

```diff
@@ -593,53 +614,20 @@ def _get_featurized_attribute(
         else:
             return [], []
 
-    def _set_attribute_features(
-        self,
-        attribute: Text,
-        sequence_features: List[scipy.sparse.spmatrix],
-        sentence_features: List[scipy.sparse.spmatrix],
-        examples: List[Message],
-    ) -> None:
-        """Set computed features of the attribute to corresponding message objects"""
-        for i, message in enumerate(examples):
-            # create bag for each example
-            if sequence_features[i] is not None:
-                final_sequence_features = Features(
-                    sequence_features[i],
-                    FEATURE_TYPE_SEQUENCE,
-                    attribute,
-                    self.component_config[FEATURIZER_CLASS_ALIAS],
-                )
-                message.add_features(final_sequence_features)
-
-            if sentence_features[i] is not None:
-                final_sentence_features = Features(
-                    sentence_features[i],
-                    FEATURE_TYPE_SENTENCE,
-                    attribute,
-                    self.component_config[FEATURIZER_CLASS_ALIAS],
-                )
-                message.add_features(final_sentence_features)
-
     def train(
-        self,
-        training_data: TrainingData,
-        cfg: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
-        """Train the featurizer.
+        self, training_data: TrainingData, spacy_nlp: Optional[SpacyModel] = None,
```

Comment:
looks like `spacy_nlp` should not be optional (or handled below)

Reply:
I made it optional as Spacy might not always be present. I'm handling the `None` case below

Reply:
oh 👀  - completely overlooked that 

another maybe stupid followup question: this is modifying `self.OOV_words` during `train` - could the config that we pass during `load` contain `OOV_words` and hence overwrite this result?

Reply:
very good point! We'll need to persist it separately

Reply:
thanks for the catch ❤️ Updated it